src/schedule/SchedulePing.ts

```diff
@@ -34,10 +40,17 @@ export class SchedulePing {
     this.logger.debug(`This schedule is ${active ? '' : 'not '}active`);
     if (active) {
       await this.schedulesRepository.ping(this.scheduleId);
-      if (!this.startedJobs) {
-        this.logger.debug(`This schedule just turned active`);
+      if (this.startJobsStatus === StartJobsStatus.notStarted) {
+        this.logger.debug('This schedule just turned active');
+        this.startJobsStatus = StartJobsStatus.inProgress;
+
         await this.startAllJobs();
-        this.startedJobs = true;
+
+        if ((this.startJobsStatus as StartJobsStatus) === StartJobsStatus.finished) {
```

Comment:
How would this be possible? There is no `await` point between checking for `notStarted` and setting the status to `inProgress`. Since this code will run single threaded on Node's main thread, I don't think we need the `finished` variant.

Reply:
Yes, you are correct. The condition for the log statement was not possible (but it gave me a slight relieve after spending half a day debugging this issue, wishing to have this condition logged 😉). I removed the unnecessary code.

Thanks for the quick review! 👍